syndesis/plugins/org.fusesource.ide.syndesis.extension.core/src/org/fusesource/ide/syndesis/extensions/core/model/SyndesisAction.java

```diff
@@ -120,34 +173,51 @@ public String getName() {
 	 * @param name the name to set
 	 */
 	public void setName(String name) {
+		firePropertyChange(PROPERTY_NAME, this.name, name);
 		this.name = name;
 	}
 	
 	/**
-	 * @return the otherProperties
+	 * @return the tags
 	 */
-	public Map<String, Object> getOtherProperties() {
-		return this.otherProperties;
+	public List<String> getTags() {
+		return this.tags;
 	}
 	
 	/**
-	 * @param otherProperties the otherProperties to set
+	 * @param tags the tags to set
 	 */
-	public void setOtherProperties(Map<String, Object> otherProperties) {
-		this.otherProperties = otherProperties;
+	public void setTags(List<String> tags) {
+		firePropertyChange(PROPERTY_TAGS, this.tags, tags);
+		this.tags = tags;
 	}
 	
 	/**
-	 * @return the tags
+	 * @return the pattern
 	 */
-	public List<String> getTags() {
-		return this.tags;
+	public String getPattern() {
+		return this.pattern;
 	}
 	
 	/**
-	 * @param tags the tags to set
+	 * @param pattern the pattern to set
 	 */
-	public void setTags(List<String> tags) {
-		this.tags = tags;
+	public void setPattern(String pattern) {
+		firePropertyChange(PROPERTY_PATTERN, this.pattern, pattern);
+		this.pattern = pattern;
+	}
+	
+	/**
+	 * @return the otherProperties
+	 */
+	public Map<String, Object> getOtherProperties() {
+		return this.otherProperties;
+	}
+	
+	/**
+	 * @param otherProperties the otherProperties to set
+	 */
+	public void setOtherProperties(Map<String, Object> otherProperties) {
+		this.otherProperties = otherProperties;
```

Comment:
why there is no firePropertyChange for OtherProperties?

Reply:
otherProperties should ideally never be used. that map simply catches those elements which are not covered by our model. But I can add a firePropertyChange anyways.

Reply:
added the fire call

Reply:
> otherProperties should ideally never be used. that map simply catches those elements which are not covered by our mode

Can you provide a javadoc comment on this attribute with this indication please?

Reply:
I'd say its pretty clear because of the annotations on the getter/setter methods but I will add some comment to the field too.

Reply:
I understood like "otherProperties" was a field of the schema initially

Reply:
nope...that is just a trap for all unmapped fields